products/sle/main.pm

```diff
@@ -72,9 +72,9 @@ sub cleanup_needles() {
     remove_desktop_needles("gnome");
     remove_desktop_needles("xfce");
     remove_desktop_needles("minimalx");
-    remove_desktop_needles("textmode");
+    remove_desktop_needles("textmode") unless check_var("REMOTE_MASTER", "ssh");
```

Comment:
a) use a proper if/unless - i.e. stick to the coding style of the rest of the file
b) I would rather define VIDEOMODE=text in your test and use that one here. The relationship to REMOTE_* feature is a bit too indirect IMO